@here/harp-mapview/lib/Utils.ts

```diff
@@ -486,6 +534,44 @@ export namespace MapViewUtils {
         );
     }
 
+    /**
+     * Extracts current camera tilt angle in radians.
+     *
+     * @param mapView The [[MapView]] in use.
+     */
+    export function extractCameraTilt(mapView: MapView): number {
+        const { camera, projection } = mapView;
+        // For planar projections the camera target point local tangent is the same
+        // at every point on the ground (ignoring terrain fluctuations), so we may
+        // simply use inverted ground normal for tilt calculation. This simplifies
+        // the more generic calculus used for spherical projections.
+        if (projection.type === ProjectionType.Planar) {
+            const lookAt: THREE.Vector3 = camera.getWorldDirection(cache.vector3[0]).normalize();
+            const normal: THREE.Vector3 = projection
+                .surfaceNormal(camera.position, cache.vector3[1])
+                .negate();
+            const cosTheta = lookAt.dot(normal);
+            return Math.acos(THREE.Math.clamp(cosTheta, -1, 1));
```

Comment:
Why this clamping? The cosine is by definition between -1 and 1.

Reply:
Probably because of floating point precision issues. I ran into the same problem where the length of the normalized vector was very slightly > 1. Which caused the dot product to be out of  [-1,1] range.

Reply:
@MPursche  -> true!